### ISSUE TITLE
New rule: Warn for unnecessary null check followed by "is type" pattern

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#5089]
I'd love an analyzer for this...

_Originally posted by @sharwell in https://github.com/dotnet/roslyn/pull/53223#discussion_r627533550_

**Comment by mavasani** [comment]
CA1508 covers majority of cases including flagging the null check in `obj is string s && s != null`. We should extend CA1508 to also flag the null check in `obj is string && obj != null`

**Comment by Youssef1313** [comment]
@mavasani A small note:

The message for CA1508 says that an expression is always true/false. This is correct for `obj is string s && s != null` where `s != null` is always true, and is correct for `obj is string && obj != null` when `obj != null` is always false.

But for `obj != null && obj is string`. None of them is always true/false. However, `obj != null` is redundant because it's done implicitly in `obj is string`.

**Comment by sharwell** [comment]
Right, this proposal was based on expressions of the form `x != null && x is T`, which would not be covered by CA1508.

**Comment by mavasani** [comment]
Seems like this will be more suited to the bucket of unnecessary IDExxxx code rules in Roslyn repo: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/unnecessary-code-rules
